internal/cluster/distribution/pke/node_pool.go

```diff
@@ -0,0 +1,40 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pke
+
+import (
+	"context"
+)
+
+type ExistingNodePool struct {
+	Name string
+}
```

Comment:
Also the stored footprint of a node pool is possibly going to be different for different providers under PKE providers so it might be a good idea to define this in the common PKE part as an interface type so each provider can supply its own stored node pool implementation.

Reply:
This might be true for other types which are aimed to be used as generic PKE types for various PKE providers.

Reply:
Or alternatively define this storage specific struct in the provider specific PKE package.

Reply:
Making an interface and provider specific structs will be created as part of future implementations.